src/language/concurrency.md

```diff
@@ -237,303 +287,272 @@ After handling the events, the isolate exits.
 
-You can use these primitives directly for more granular
-control over isolate functionality. For example, `run()` shuts
-down its isolate after returning a single message. 
-What if you want to allow multiple messages to pass between isolates?
-You can set up your own isolate much the same way `run()` is implemented,
-just utilizing the [`send()` method][] of `SendPort` in a slightly different way.
+In some special cases, you might need to use [`Isolate.spawnUri()`][], which
+sets up the new isolate with a copy of the code that's at the specified URI.
+However, `spawnUri()` is much slower than `spawn()`, and the new isolate isn't
+in its spawner's isolate group. Another performance consequence is that message
+passing is slower when isolates are in different groups.
 
-One common pattern, which the following figure shows,
-is for the main isolate to send a request message to the worker isolate,
-which then sends one or more reply messages.
-
-![A figure showing the main isolate spawning the isolate and then sending a request message, which the worker isolate responds to with a reply message; two request-reply cycles are shown](/assets/img/language/concurrency/isolate-custom-bg-worker.png)
+[`Isolate.spawnUri()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawnUri.html
 
-Check out the [long_running_isolate.dart][] sample,
-which shows how to spawn a long-running isolate
-that receives and sends messages multiple times between isolates.
+### Limitations of isolates
 
-{% assign samples = "https://github.com/dart-lang/samples/tree/main/isolates" %}
+#### Isolates aren't threads
 
-[isolate samples]: {{ samples }}
-[send_and_receive.dart]: {{ samples }}/bin/send_and_receive.dart
-[long_running_isolate.dart]: {{ samples }}/bin/long_running_isolate.dart
+If you’re coming to Dart from a language with multithreading, it’d be reasonable
+to expect isolates to behave like threads, but that isn’t the case. Each isolate
+has its own memory heap, ensuring that none of the state in an isolate is
+accessible from any other isolate. Therefore, isolates are limited by their
+access to their own memory. 
 
+For example, if you have an application with a
+global mutable variable, that variable will be a separate
+variable in your spawned isolate. If you mutate that variable in the spawned
+isolate, it will remain untouched in the main isolate. This is how isolates are
+meant to function, and it's important to keep in mind when you’re considering
+using isolates.
 
-## Performance and isolate groups
+#### Message types
 
-When an isolate calls [`Isolate.spawn()`][],
-the two isolates have the same executable code
-and are in the same _isolate group_.
-Isolate groups enable performance optimizations such as sharing code;
-a new isolate immediately runs the code owned by the isolate group.
-Also, `Isolate.exit()` works only when the isolates
-are in the same isolate group.
+Messages sent via [`SendPort`][]
```

Comment:
Do consider mentioning `Isolate.run` and `Isolate.exit` here as it may not be clear to readers that they are subject to the same limitations.